File: bassplayerlib/src/main/java/ru/sigil/bassplayerlib/listeners/IStreamChangedListener.java
package ru.sigil.bassplayerlib.listeners;

import ru.sigil.bassplayerlib.IRadioStream;

/**
 * Created by namelessone
 * on 10.06.17.
 */

public interface IStreamChangedListener <T extends IRadioStream> {
    //TODO generic всё портит=(
    void onStreamChanged(T stream);
}

File: bassplayerlib/src/main/kotlin/ru/sigil/bassplayerlib/listeners/IStreamChangedListener.kt
package ru.sigil.bassplayerlib.listeners

import ru.sigil.bassplayerlib.IRadioStream

/**
 * Created by namelessone
 * on 15.12.18.
 */
interface IStreamChangedListener<T : IRadioStream> {
    fun onStreamChanged(stream: T?)
}